flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResultPartition.java

```diff
@@ -93,21 +89,12 @@ void resetForNewExecution() {
         hasDataProduced = false;
     }
 
-    int addConsumerGroup() {
-        int pos = consumers.size();
-
-        // NOTE: currently we support only one consumer per result!!!
-        if (pos != 0) {
-            throw new RuntimeException(
-                    "Currently, each intermediate result can only have one consumer.");
-        }
-
-        consumers.add(new ArrayList<ExecutionEdge>());
-        return pos;
+    public void addConsumers(ConsumerVertexGroup consumers) {
+        getEdgeManager().addPartitionConsumers(partitionId, consumers);
     }
 
-    void addConsumer(ExecutionEdge edge, int consumerNumber) {
-        consumers.get(consumerNumber).add(edge);
+    private EdgeManager getEdgeManager() {
+        return producer.getExecutionGraph().getEdgeManager();
     }
```

Comment:
I think this shows that we are overly coupling the `IntermediateResultPartition` with the `ExecutionGraph`. Couldn't we give an `EdgeManager` to the `IntermediateResultPartition` when we create it? This makes the dependency explicit.

Reply:
Theoretically yes. But the `ExecutionVertex` is already coupled quite tightly to the `ExecutionGraph`. Hence, it might not make a big difference to not pass it in.